packages/clerk-js/src/ui/router/BaseRouter.tsx

```diff
@@ -121,6 +121,10 @@ export const BaseRouter = ({
     return internalNavRes;
   };
 
+  React.useEffect(() => {
+    return __internal_setComponentNavigationContext?.({ basePath, navigate: baseNavigate });
+  }, []);
```

Comment:
This might be better suited in our actual components, if `BaseRouter` isn't getting unmounted, per your other comment!

Reply:
> This might be better suited in our actual components

My first implementation was only calling the following on `SignInContext/SignUpContext` - without a cleanup: 

```
React.useEffect(() => {
     __internal_setComponentNavigationContext?.({ basePath, navigate: baseNavigate });
  }, []);
```

The reason why, is that from `_SignInStart` to `_SignInFactorOne`, the cleanup gets triggered and we lose the reference before executing `setActive` 

Before, I was placing the effect on two different places (the component contexts) and then executing the cleanup after the navigation - which felt disconnected, and then I tried to move the setter at least to a single place

Reply:
@LauraBeatris can we put it in the `<SignIn>` component itself then?

Reply:
Putting inside each root component works better than my previous approach of relying on their React context! 

Here's the refactor (take a look at the git description for more details on the other changes as well) 🫡 https://github.com/clerk/javascript/pull/5280/commits/57203a1e1ceeb78869f8ceb3c61296bedcef8102